File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task38_PERPLEXITY_llama-3-sonar-large-32k-chat.c
#include <stdio.h>

int main() {
    printf("Content-type: text/html\r
\r
");
    printf("<html><body>");
    if (getenv("REQUEST_METHOD") && strcmp(getenv("REQUEST_METHOD"), "POST") == 0) {
        char name[1024];
        scanf("%s", name);
        printf("<h1>Hello, %s!</h1>", name);
    } else {
        printf("<form action=\"\" method=\"post\">");
        printf("    <input type=\"text\" name=\"name\" placeholder=\"Enter your name\">");
        printf("    <input type=\"submit\" value=\"Submit\">");
        printf("</form>");
    }
    printf("</body></html>");
    return 0;
}

File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task38_PERPLEXITY_llama-3-sonar-large-32k-chat.java
import javax.servlet.http.*;
import java.io.*;

public class HelloWorldServlet extends HttpServlet {
    public void doPost(HttpServletRequest request, HttpServletResponse response)
            throws IOException, ServletException {
        String name = request.getParameter("name");
        response.setContentType("text/html");
        PrintWriter out = response.getWriter();
        out.println("<h1>Hello, " + name + "!</h1>");
    }

    public void doGet(HttpServletRequest request, HttpServletResponse response)
            throws IOException, ServletException {
        response.setContentType("text/html");
        PrintWriter out = response.getWriter();
        out.println("<form action=\"\" method=\"post\">");
        out.println("    <input type=\"text\" name=\"name\" placeholder=\"Enter your name\">");
        out.println("    <input type=\"submit\" value=\"Submit\">");
        out.println("</form>");
    }
}